src/Elastic.Apm.AspNetCore/ApmMiddleware.cs

```diff
@@ -34,21 +33,23 @@ internal class ApmMiddleware
 
 		private readonly RequestDelegate _next;
 		private readonly Tracer _tracer;
-		private readonly IConfigurationReader _configurationReader;
 		private readonly ApmAgent _agent;
 
 		public ApmMiddleware(RequestDelegate next, Tracer tracer, ApmAgent agent)
 		{
 			_next = next;
 			_tracer = tracer;
 			_logger = agent.Logger.Scoped(nameof(ApmMiddleware));
-			_configurationReader = agent.ConfigurationReader;
 			_agent = agent;
 		}
 
 		public async Task InvokeAsync(HttpContext context)
 		{
-			var transaction = WebRequestTransactionCreator.StartTransactionAsync(context, _logger, _tracer, _agent.ConfigStore.CurrentSnapshot);
+			var newTransaction = WebRequestTransactionCreator.StartTransactionAsync(context, _logger, _tracer, _agent.ConfigStore.CurrentSnapshot);

```

Comment:
Maybe handle the result of `WebRequestTransactionCreator.StartTransactionAsync` as `Transaction`? e.g.

```c#
var transaction = WebRequestTransactionCreator.StartTransactionAsync(context, _logger, _tracer, _agent.ConfigStore.CurrentSnapshot) as Transaction;
```

`newTransaction` is not used unless it's a `Transaction`, so this may simplify the flow a little

Reply:
This makes sense and I also started changing it accordingly, but then I realized there was a bug in this part:

We need the `ITransaction` returned there, because we still maintain `Tracer.CurrentTransaction` when `recording=false`, so we also need to call `ITtransaction.End()` when the request is finished. So I think there is no way to immediately cast it to `Span`, because that way we won't be able to call `End()` on the created `NoopTransaction`.